security_privacy_ln.asciidoc

```diff
@@ -0,0 +1,614 @@
+This well-known search technique is called _binary search_.
+With binary search, the number of probes is _logarithmic_ in the desired precision.
+For example, to obtain Alice's balance in a channel of 1 million satoshis up to a single satoshi, Mallory would only have to perform stem:[\log_2(1000000) \approx 20] probings.
+If one probing takes 3 seconds, one channel can be precisely probed in only about a minute!
+
+Channel probing can be made even more efficient.
+In its simplest variant, Mallory directly connects to the channel she wants to probe.
+Is it possible to probe a channel without opening a channel to one of its endpoints?
+Imagine Mallory now wants to probe a channel between Bob and Charlie but doesn't want to open another channel, which requires paying on-chain fees and waiting for confirmations of the funding transactions.
+Instead, Mallory re-uses her existing channel to Alice and sends a probe along the route Mallory - Alice - Bob - Charlie.
+Mallory can interpret the "unknown payment hash" error in the same way as before: the probe has reached the destination; therefore, all channels along the route have sufficient balances to forward it.
+But what if Mallory receives the "insufficient balance" error?
+Does it mean that the balance is insufficient between Alice and Bob or between Bob and Charlie?
+
+In the current Lightning protocol, error messages report not only _which_ error occurred but also _where_ it happened.
+So, with more careful error handling, Mallory now knows which channel failed.
+If this is the target channel, she updates her estimates; if not, she chooses another route to the target channel.
+She even gets _additional_ information about the balances of intermediary channels, on top of that of the target channel.
+
+The probing attack can be further used to link senders and receivers, as described in the previous section.
+
+At this point, you may ask: why does the Lightning Network do such a poor job at protecting its users' private data?
+Wouldn't it be better to not reveal to the sender why and where the payment has failed?
+Indeed, this could be a potential countermeasure, but it has significant drawbacks.
+Lightning has to strike a careful balance between privacy and efficiency.
+Remember that regular nodes don't know balance distributions in remote channels.
+Therefore, payments can (and often do) fail because of insufficient balance at an intermediary hop.
+Error messages allow the sender to exclude the failing channel from consideration when constructing another route.
+A popular Lightning wallet Zap even performs probing internally to check whether a constructed route can really handle a payment.
+
+There are other potential countermeasures against channel probing.
+First, it is hard for an attacker to target unannounced channels.
+Second, nodes that implement JIT routing are less prone to the attack.
```

Comment:
I think that should be relaxed to "may be" less prone to probing, as to my knowledge it hasn't really been deployed at a large scale to be able to draw solid empirical conclusions. It seems as though JIT routing would also exacerbate the issue of HTLC slot utilization, given that a single prob across `N` nodes would then cause each of those nodes to attempt to generate up to `K` routes themselves to attempt to rebalancing things, each of which may actually fail.